### ISSUE TITLE
[clang] Incorrect type error when using a variable member template of class template

### ISSUE THREAD

**Issue by wreien** [llvm/llvm-project#125181]
Consider the following valid C++14 snippet (https://godbolt.org/z/jYGfzTf7P):
```cpp
template <typename>
struct S {
  template <typename> static const auto x = 1;
};

int r = S<int>::x<int>;
```
Clang trunk gives the following error:
```
<source>:6:5: error: cannot initialize a variable of type 'int' with an lvalue of type 'auto'
    6 | int r = S<int>::x<int>;
      |     ^   ~~~~~~~~~~~~~~
```
Removing either template resolves the error.  Marking `x` as `constexpr` also fixes in this particular case.

This error appears to have started with Clang 9.

**Comment by shafik** [comment]
This straight up just looks like a bug that started around clang-8: https://godbolt.org/z/7adcd4Gxq

**Comment by cor3ntin** [comment]
@mizvekov
